eclipse/che-theia
# Serve mini-browser endpoint on a dedicated route

Signed-off-by: Artem Zatsarynnyi <azatsary@redhat.com>



_Note: It adapts Che Theia to the upstream changes.
Once the upstream PR is merged this PR should be switched to build from Theia master branch.
See the details in https://github.com/eclipse/che/issues/18456
`CI / node-build` PR checks should pass once the corresponding npm packages are published to npmjs._

### What does this PR do?
Makes Che Theia serving mini-browser endpoint on a dedicated route to restrict access to main Che Theia page from a page loaded into iframe through mini-browser endpoint.
The corresponding Che endpoint has been exposed in the PR https://github.com/eclipse/che-plugin-registry/pull/712

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

closes https://github.com/eclipse/che/issues/18456

### How to test this PR?


1. Spin-up a Workspace using one of the `Try it` buttons posted by @che-bot.
2. Open any HTML file in Preview mode.
3. Using browser's DevTools make sure mini-browser iframe content is loaded from a dedicated host that differs from the che-theia host.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### Happy Path Channel

HAPPY_PATH_CHANNEL=stable
